autoload/maktaba/syscall.vim

```diff
@@ -67,6 +69,24 @@ function! s:DoSyscallCommon(syscall, CallFunc, throw_errors) abort
 endfunction
 
 
+" Execute a function in given tab, used for callbacks of asynchronous actions.
+function! s:CallbackInTab(callback, source_tab) abort
+  call maktaba#ensure#IsCallable(a:callback)
+  call maktaba#ensure#IsNumber(a:source_tab)
+  let l:current_tab = tabpagenr()
+  let l:state = maktaba#value#Save('lazyredraw')
+  let &lazyredraw = 1
+  try
+    silent! execute "tabnext " . a:source_tab
```

Comment:
I don't think `:tabnext` has any error output, even if you give it an invalid tab number, so the `:silent!` seems like overkill. Can you switch to
```vim
execute 'tabnext' a:source_tab
```
and similar below?